### PR TITLE
Snackbar

### PR DESCRIPTION
**Pull request by vanshBhatia-A4k9**
Added the Snackbar component

Screenshot from 2021-07-05 12-50-50
Screenshot from 2021-07-05 12-50-59
Screenshot from 2021-07-05 12-51-02
Screenshot from 2021-07-05 13-11-30

### CODE REVIEW

File: src/core/Snackbar/Snackbar.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import IconButton from "@material-ui/core/IconButton";
+import { SnackbarProps as SnackbarBaseProps } from "@material-ui/core/Snackbar";
+import CloseIcon from "@material-ui/icons/Close";
+import React from "react";
+import { useStyles } from "./styles";
+
+type Variant = "success" | "warning" | "error" | undefined;
+
+export interface SnackbarProps extends SnackbarBaseProps {
+  variant?: Variant;
```
Comment (whitetiger1399) [inline src/core/Snackbar/Snackbar.tsx:11]: May you handle the styling for the case of `variant: undefined`. Either assign some styling as a fallback or make any of the existing options as default

**Review by whitetiger1399 — CHANGES_REQUESTED** [review]
Please check the variant prop for undefined case

**Review by whitetiger1399 — APPROVED** [review]
Awesome work!! Thanks a million

File: src/core/Snackbar/__tests__/Snackbar.test.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    <LitmusThemeProvider>
+      <Snackbar message={"Test Snackbar"} open={true} />
+    </LitmusThemeProvider>
+  );
+});
+
+test("Check for inheritance of children props", () => {
+  // Matching a string:
+  screen.getByText("Test Snackbar"); // full string match
+  screen.getByText("est Snack", { exact: false }); // substring match
```
Comment (arkajyotiMukherjee) [inline src/core/Snackbar/__tests__/Snackbar.test.tsx]: This test how is it helping

File: src/core/Snackbar/__tests__/Snackbar.test.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  screen.getByText("tEsT sNaCkBaR", { exact: false }); // ignore case
+
+  // Matching a regex:
+  screen.getByText(/Snackbar/); // substring match
+  screen.getByText(/sNaCkBaR/i); // substring match, ignore case
```
Comment (arkajyotiMukherjee) [inline src/core/Snackbar/__tests__/Snackbar.test.tsx]: these as well...We should be only checking full string in the exact case as it is, right?

File: src/core/Snackbar/__tests__/Snackbar.test.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  screen.getByText(/^tEsT sNaCkBaR$/i); // full string match, ignore case
+
+  // Matching with a custom function:
+  screen.getByText((content, element) => content.startsWith("Test"));
```
Comment (arkajyotiMukherjee) [inline src/core/Snackbar/__tests__/Snackbar.test.tsx]: same

File: src/core/Snackbar/__tests__/Snackbar.test.tsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  // Case-sensitive regex with different case
+  expect(() => {
+    screen.getByText(/test snackbar/);
+  }).toThrow();
+
+  // Function looking for a paragraph when it's actually a span:
+  expect(() => {
+    screen.getByText((content, element) => {
+      return (
+        element?.tagName.toLowerCase() === "p" && content.startsWith("Test")
```
Comment (arkajyotiMukherjee) [inline src/core/Snackbar/__tests__/Snackbar.test.tsx]: This is internal with Mui...do we need to check this

File: src/core/Snackbar/__tests__/Snackbar.test.tsx [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+  render(
+    <LitmusThemeProvider>
+      <Snackbar message={"Test Snackbar"} open={true} />
+    </LitmusThemeProvider>
+  );
+});
+
+test("Check for inheritance of children props", () => {
+  // Matching a string:
+  screen.getByText("Test Snackbar"); // full string match
```
Comment (arkajyotiMukherjee) [inline src/core/Snackbar/__tests__/Snackbar.test.tsx:16]: This test is good

**Review by arkajyotiMukherjee — CHANGES_REQUESTED** [review]
Some testcases I feel is not actually testing the component...those can be removed @vanshBhatia-A4k9

**Review by arkajyotiMukherjee — APPROVED** [review]

**Pull request merged**